src/Polygon.js

```diff
@@ -12,7 +15,14 @@ export default class Polygon extends PopupContainer {
 
   componentDidUpdate(prevProps) {
     if (this.props.positions !== prevProps.positions) {
-      this.leafletElement.setLatLngs(this.props.positions);
+      if(!isEqual(this.props.positions, prevProps.positions)){
```

Comment:
Well that was the whole point. Say in a render method you'll do something like this
``` javascript
var positions = _.map(this.props.geoJsonPoints, (point)->{  return [point[1], point[0]})
// or some other transformation of props that result in a new positions array
render <Polygon positions={positions} />
```
than you're constructing a new positions array on each render and `!==` does not wor. You are re-rendering the polygon even its positions didn't change (leaflet doesn't check that internally either) ie. touching the DOM which is bad for performance and very bad if you are re-rendering often especially polygons.
You can obviously work around that in your render method but if you're not aware of it its easy to miss. So this is just a react'y way of not touching the DOM if we don't have to.